go-controller/pkg/factory/types.go

```diff
@@ -29,6 +29,7 @@ type NodeWatchFactory interface {
 	Shutdownable
 
 	AddServiceHandler(handlerFuncs cache.ResourceEventHandler, processExisting func([]interface{})) *Handler
+	AddFilteredServiceHandler(namespace string, matchSel labels.Selector, handlerFuncs cache.ResourceEventHandler, processExisting func([]interface{})) *Handler
```

Comment:
as previous comment, if this is a oneshot thing, I think is better to use the `AddServiceHandler` directly and do the filtering there,